source/extensions/filters/http/jwt_authn/jwks.cc

```diff
@@ -0,0 +1,254 @@
+#include "extensions/filters/http/jwt_authn/jwks.h"
+
+#include "common/common/assert.h"
+#include "common/common/base64.h"
+#include "common/json/json_loader.h"
+
+#include "extensions/filters/http/jwt_authn/utils.h"
+
+#include "openssl/bn.h"
+#include "openssl/ecdsa.h"
+#include "openssl/evp.h"
+#include "openssl/rsa.h"
+#include "openssl/sha.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+namespace {
+
+/** Class to create EVP_PKEY object from string of public key, formatted in PEM or JWKs.
+ * If it failed, status_ holds the failure reason.
+ *
+ * Usage example:
+ * EvpPkeyGetter e;
+ * bssl::UniquePtr<EVP_PKEY> pkey = e.createEvpPkeyFromStr(pem_formatted_public_key);
+ * (You can use createEvpPkeyFromJwkRSA() or createEcKeyFromJwkEC() for JWKs)
+ */
+class EvpPkeyGetter : public WithStatus {
+public:
+  // Create EVP_PKEY from PEM string
+  bssl::UniquePtr<EVP_PKEY> createEvpPkeyFromStr(const std::string& pkey_pem) {
+    std::string pkey_der = Base64::decode(pkey_pem);
+    if (pkey_der == "") {
+      updateStatus(Status::JwksPemBadBase64);
+      return nullptr;
+    }
+    auto rsa =
+        bssl::UniquePtr<RSA>(RSA_public_key_from_bytes(castToUChar(pkey_der), pkey_der.length()));
+    if (!rsa) {
+      updateStatus(Status::JwksPemParseError);
+    }
```

Comment:
You fall through here, and then do a `rsa.get()`. Is that what you want to do? Do you have test coverage of this path?